public/app/plugins/datasource/jaeger/datasource.ts

```diff
@@ -146,3 +123,66 @@ function getTime(date: string | DateTime, roundUp: boolean) {
   }
   return date.valueOf() * 1000;
 }
+
+function createTraceFrame(data: TraceResponse): DataFrame {
+  const spans = data.spans.map((s) => toSpanRow(s, data.processes));
+
+  return new MutableDataFrame({
+    fields: mapFields(spans),
+    meta: {
+      preferredVisualisationType: 'trace',
+    },
+  });
+}
+
+function mapFields(data: Array<Record<string, any>>): FieldDTO[] {
+  const map = data.reduce((acc, datum) => {
+    for (const key of Object.keys(datum)) {
+      if (!acc[key]) {
+        acc[key] = {
+          name: key,
+          type: guessFieldType(datum[key]),
+          values: [],
+        };
+      }
+      (acc[key].values! as any[]).push(datum[key as keyof typeof datum]);
+    }
+    return acc;
+  }, {} as Record<string, FieldDTO>);
+  return Object.values(map);
+}
```

Comment:
Maybe comment that describes the structure of the data going in, and out. Find it a bit hard to read and reason about this code (reduce always messing with my mind).  But after reading it more a couple of times it takes in an array of objects and returns an array of fields for each property on the object?

Reply:
Yeah it's just transforms array of rows to an array of fields with values for the data frame. Will check cause maybe we already have a way to do that transform some where.